extensions/azurecore/src/azureResource/providers/sqlinstanceArc/sqlInstanceArcTreeDataProvider.ts

```diff
@@ -0,0 +1,78 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { ExtensionNodeType, TreeItem } from 'azdata';
+import { TreeItemCollapsibleState, ExtensionContext } from 'vscode';
+import * as nls from 'vscode-nls';
+const localize = nls.loadMessageBundle();
+
+import { AzureResourceItemType } from '../../constants';
+import { ApiWrapper } from '../../../apiWrapper';
+import { generateGuid } from '../../utils';
+import { IAzureResourceService } from '../../interfaces';
+import { ResourceTreeDataProviderBase } from '../resourceTreeDataProviderBase';
+import { azureResource } from '../../azure-resource';
+
+export class SqlInstanceArcTreeDataProvider extends ResourceTreeDataProviderBase<azureResource.AzureResourceDatabaseServer> {
+	private static readonly containerId = 'azure.resource.providers.sqlInstanceArcContainer';
+	private static readonly containerLabel = localize('azure.resource.providers.sqlInstanceArcContainerLabel', "Azure SQL DB managed instance – Azure Arc");
+
+	public constructor(
+		databaseServerService: IAzureResourceService<azureResource.AzureResourceDatabaseServer>,
+		apiWrapper: ApiWrapper,
+		private _extensionContext: ExtensionContext
+	) {
+		super(databaseServerService, apiWrapper);
+	}
+
+
+	protected getTreeItemForResource(databaseServer: azureResource.AzureResourceDatabaseServer): TreeItem {
+		return {
+			id: `sqlInstance_${databaseServer.id ? databaseServer.id : databaseServer.name}`,
+			label: databaseServer.name,
+			iconPath: {
+				dark: this._extensionContext.asAbsolutePath('resources/dark/sql_instance_inverse.svg'),
+				light: this._extensionContext.asAbsolutePath('resources/light/sql_instance.svg')
+			},
+			collapsibleState: TreeItemCollapsibleState.Collapsed,
+			contextValue: AzureResourceItemType.databaseServer,
+			payload: {
+				id: generateGuid(),
+				connectionName: undefined,
+				serverName: databaseServer.fullName,
+				databaseName: databaseServer.defaultDatabaseName,
+				userName: databaseServer.loginName,
+				password: '',
+				authenticationType: 'SqlLogin',
```

Comment:
only SqlLogin is supported here?

Reply:
This seems to be mostly broken currently (I just copied what Kevin had which was mostly for demo purposes).

I'm going to add the preview tag to the setting so it's clear this is still in development and then we'll be working to improve this stuff later on.

Reply:
This is probably just copied from what we do for azure sql resources currently. This is the spot where working with @aaomidi would be good to, if we have the token for doing aad, specifying this login type as aad instead.